File: src/Services/SalesService/SalesService.Domain/Entities/Product.cs
using NanoidDotNet;

namespace SalesService.Domain.Entities
{
    public class Product
    {
        public string Id { get; private set; } = Nanoid.Generate(size: 21);
        public string CompanyId { get; set; } = string.Empty;
        public string Name { get; set; } = string.Empty;
        public string Description { get; set; } = string.Empty;
        public string Category { get; set; } = string.Empty;
        public decimal Price { get; set; } = 0;
        public int Tax { get; set; } = 0;
        public DateTime CreatedAt { get; private set; } = DateTime.UtcNow;
    }
}

File: src/WebApp/smartbusiness.web/src/models/product.ts
export interface Product {
  id: string;
  companyId: string;
  name: string;
  description: string;
  category: string;
  price: number;
  tax: number;
  createdAt: string; // ISO string (DateTime)
}

export type NewProduct = Omit<Product, "id" | "companyId" | "createdAt">;

export interface ProductContextType {
  createProduct: (product: NewProduct) => Promise<void>;
  updateProduct: (product: Partial<Product>) => Promise<void>;
  deleteProduct: (productId: string) => Promise<void>;
  fetchProduct: (productId: string) => Promise<Product | null>;
  fetchProducts: (params: GetProductsByParamsQuery) => Promise<Product[]>;
}

export interface GetProductsByParamsQuery {
  Name?: string;
  Category?: string;
  MinPrice?: number;
  MaxPrice?: number;
}